### PR TITLE
[Merged by Bors] - feat: Handle powers and multiplicative inverses in cancel_denoms

### LINKED ISSUE #7732: `linarith` Cannot Handle `2⁻¹`
Hi,

I just found that `linarith` can solve
```
x: ℝ
h₀: x * (1 / 2 + 2 / 3) = 1
⊢ x = 6 / 7
```
but cannot solve
```
x: ℝ
h₀: x * (2⁻¹ + 2 / 3) = 1
⊢ x = 6 / 7
```

Would it be possible/desirable for `linarith` to recognize that `2⁻¹` is also in the linear arithmetic fragment?

More context: This behavior has prevented `aesop` from proving some theorems (see https://github.com/JLimperg/aesop/issues/76). The second goal is the result of applying `simp_all` to the first goal, and `simp_all` is used extensively in aesop for normalizing proof goals.

### PR DESCRIPTION
**Pull request by sebzim4500**
fixes #7732 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

Also solves the specific example in #7808, but I would imagine that there are still cases where `cancel_denoms` is not idempotent.

### CODE REVIEW

**Comment by grunweg** [comment]
Drive-by comment: I think the proper way to format things (so they can be picked up by github) is to make "feat: Handle powers and multiplicative inverses in cancel_denoms" the PR title. The "fixes ..." part should go above the line, so it will become part of the eventual commit message.

Thanks for doing this, I'm always excited to see tactic improvements!

**Comment by robertylewis** [comment]
This is great -- thanks for the quick fix!

bors merge

File: Mathlib/Tactic/CancelDenoms/Core.lean [hunk 1]
```diff
@@ -177,6 +197,24 @@ partial def mkProdPrf (α : Q(Type u)) (sα : Q(Field $α)) (v : ℕ) (t : Tree
   | t, ~q(-$e) => do
     let v ← mkProdPrf α sα v t e
     mkAppM ``CancelDenoms.neg_subst #[v]
+  | .node _ lhs@(.node k1 _ _) (.node k2 .nil .nil), ~q($e1 ^ $e2) => do
```
Comment (eric-wieser) [inline Mathlib/Tactic/CancelDenoms/Core.lean:200]: Probably better as
```suggestion
  | .node _ lhs@(.node k1 _ _) (.node k2 .nil .nil), ~q($e1 ^ ($e2 : ℕ)) => do
```

**Review by eric-wieser — COMMENTED** [review]
It would be cool to handle `zpow` here too, such as `2 ^ (-1)`; but it's very reasonable to dexlare that out of scope.

**Pull request closed without merging**